### DISCUSSION TITLE
Persistence and durability

### DISCUSSION THREAD

**Discussion by WalManas** [rabbitmq/rabbitmq-server#5119] [Other]
Hello,

I have a couple of questions about RabbitMQ durability.

First, at this [link](https://www.rabbitmq.com/queues.html#:~:text=Persistent%20messages%20routed%20to%20durable%20queues%20are%20persisted%20in%20batches%20or%20when%20a%20certain%20amount%20of%20time%20passes%20(fraction%20of%20a%20second).), there a sentence that says: "Persistent messages routed to durable queues are persisted in batches or when a certain amount of time passes (fraction of a second)."
Can I set the batch size? What it the parameter used to do so?
Can I set the amount of time before messages are persisted?  What it the parameter used to do so?

My second concern is that, is there is any configuration parameter that entirely disable persistence? Either by disabling the creation of durable queues, or by entirely disabling the server from writing to disk?

Thanks in advance,

Wael.

**Comment by michaelklishin** [comment]
That page will likely soon enough see a round of major changes. This is an implementation detail, and in 3.10, CQv2 would different from CQv1, and QQs and streams are different from both of those.

There are no parameters that disable persistence. If anything, RabbitMQ has been moving towards making this optional durability idea of AMQP 0-9-1 a thing of the past almost everywhere (except for classic non-mirrored queues, perhaps). Raft requires persistent storage, for example, so do QQs and streams.

You can run RabbitMQ on a disk partition that is backed RAM but I would definitely not recommend such setups. Nodes would not even know who its previous cluster members were upon reboot.

**Comment by johanrhodin** [comment]
I see that this was double-posted to `rabbitmq-users`. Adding my reply here for completeness:

1. & 2. If it is possible I would recommend against it (wrong parameter to try and tune for very little benefit) and instead look at publisher confirms and how many outstanding confirms you have: https://www.rabbitmq.com/confirms.html#when-publishes-are-confirmed
 
Disable persistence? You can send transient messages to non-durable queues, but I don't think you can enforce it. And even non-durable queues will page out transient messages to disk when memory usage calls for it.
